File: miscellaneous/graph-bfs/20230722/graph-bfs.php
<?php

function graphBfs(array $adjList): array
{
    $queue = [0];
    $seen = [0];
    $values = [];

    while ($queue) {
        $current = array_shift($queue);
        $values[] = $current;
        $connections = $adjList[$current];

        foreach ($connections as $connection) {
            if (!in_array($connection, $seen)) {
                $seen[] = $connection;
                $queue[] = $connection;
            }
        }
    }

    return $values;
}

File: miscellaneous/graph-bfs/20230722/graph_bfs.py
from typing import List, Tuple


def bfs(adj_list: Tuple, start: int) -> List:
    queue = [start]
    seen = {start}
    results: [int] = []

    while queue:
        current = queue.pop(0)
        connections = adj_list[current]
        results.append(current)

        for connection in connections:
            if connection not in seen:
                queue.append(connection)
                seen.add(connection)

    return results
